connect-n.js

```diff
@@ -22,21 +24,87 @@ function createArray(w, h) {
 
 const board = createArray(width, height);
 
+// Add an index row to the bottom of the board so users know where to drop their stones
+const indexRow = [];
+for (let i = 0; i < width; i++) {
+  indexRow[i] = (i + 1).toString();
+}
+board.push(indexRow);
+
 function printBoard(array) {
-  const indexRow = [];
-  for (let i = 0; i < width; i++) {
-    indexRow[i] = (i + 1).toString();
+  array.forEach((row) => {
+    // The table will only look good if there are fewer than 10 columns
+    // TODO: use a fancy curses library or something to make tables look good
+    console.log(`| ${row.join(' | ')} |`.replace(/ {2}/g, ' . '));
+  });
+}
+
+function testColumn(index, playerSymbol) {
+  const result = {};
+  if (board[0][index]) {
+    console.log('Column is full, try another one');
+    return false;
   }
-  array.push(indexRow);
+  for (let i = 0; i < height; i++) {
+    if (!board[i][index]) {
+      result.x = index;
+      result.y = i;
+    } else {
+      board[i - 1][index] = playerSymbol;
+      return result;
+    }
+  }
+  board[height - 1][index] = playerSymbol;
+  return result;
+}
 
-  array.forEach((row) => {
-    for (let i = 0; i < width; i++) {
-      if (row[i] === undefined) {
-        row[i] = '.';
-      }
+async function dropStone(playerSymbol) {
+  let coord;
+  while (!coord) {
+    const schema = {
+      properties: {
+        column: {
+          description: 'Enter the column you wish to drop your stone in',
+          pattern: '^[1-9]\\d*$',
+          message: 'Must be a positive integer',
+          required: true,
+        },
+      },
+    };
+    await prompt.start();
+    const { column } = await prompt.get(schema);
+    coord = testColumn(column - 1, playerSymbol);
+  }
+
+  return coord;
+}
+
+async function gameLoop() {
+  let result;
+  while (!result) {
```

Comment:
Right now, there might be a slight delay in seeing the result here, as it will only be checked each time the `while` condition is checked - meaning it will need to iterate fully through the inner `for` loop each time.
Perhaps rather than having two nested loops (`while` and `for`), it might be simpler to just use one, with an `if` to check for that condition and `break` when we get a result.

i.e. something like:

```
  let result;
  for (let counter = 0; counter < width * height; counter++) {
    if (result) {
      break;
    }
    ...
  }
```

Reply:
Done